### PR TITLE
Disabled index pattern checks in Healthcheck

### LINKED ISSUE #3251: Error checking tempalte and pattern fields when index pattern check is disabled
| Wazuh | Elastic | Rev |
| ----- | ------- | --- |
| 4.x   | 7.x     | 4x |

**Description**
When you disable the pattern check the health check failed. The template pattern and the pattern fields should not be performed due to dependency with the pattern check.

**Steps to reproduce**

1. Disable check pattern from wazuh.yml
2. Run the health-check
3. See the error

**Screenshots**
Screenshot_20210510_155405

Check with the cloud team the option to make required this check.

### PR DESCRIPTION
**Pull request by Machi3mfl**
Hi Team,
This PR adds and fixes:

- Throw error when `checks.pattern` is disable in wazuh.yml and all index patterns not exists.
- Remove automatic default index patterns creation if `checks.pattern` is `false`.
- Default index pattern are created if `checks.pattern` is `true`.
- Combine Pattern, Fields, Template Checks in one `Index pattern Check`. 

Closes #3251

### CODE REVIEW

**Comment by Machi3mfl** [comment]
To reproduce:
- Crear cache storage
- Remove all index patterns

Screen Shot 2021-05-28 at 16 34 39

File: public/components/health-check/container/health-check.container.tsx [hunk 1]
```diff
@@ -67,39 +65,26 @@ const checks = {
...
-    awaitFor: ["pattern"],
+    shouldCheck: true,
     canRetry: true,
   },
   patternMonitoring: {
     title: 'Check monitoring index pattern',
     label: 'Monitoring index pattern',
     validator: (appConfig) => checkPatternSupportService(appConfig.data['wazuh.monitoring.pattern'], WAZUH_INDEX_TYPE_MONITORING),
-    awaitFor: [],
+    awaitFor: ["pattern"],
```
Comment (Desvelao) [inline public/components/health-check/container/health-check.container.tsx:79]: Why this check is waiting for the check of alert index pattern? These are not related.

File: public/components/health-check/container/health-check.container.tsx [hunk 2]
```diff
@@ -67,39 +65,26 @@ const checks = {
...
     shouldCheck: true,
     canRetry: true,
   },
   patternStatistics: {
     title: 'Check statistics index pattern',
     label: 'Statistics index pattern',
     validator: (appConfig) => checkPatternSupportService(`${appConfig.data['cron.prefix']}-${appConfig.data['cron.statistics.index.name']}-*`, WAZUH_INDEX_TYPE_STATISTICS),
-    awaitFor: [],
+    awaitFor: ["pattern"],
```
Comment (Desvelao) [inline public/components/health-check/container/health-check.container.tsx:87]: Why this check is waiting for the check of alert index pattern? These are not related.

File: public/components/health-check/container/health-check.container.tsx [hunk 3]
```diff
@@ -67,39 +65,26 @@ const checks = {
     awaitFor: ["api"],
   },
   pattern: {
-    title: 'Check alerts index pattern',
-    label: 'Alerts index pattern',
+    title: 'Check index pattern',
+    label: 'Index pattern',
```
Comment (Desvelao) [inline public/components/health-check/container/health-check.container.tsx:69]: The `alert` word helps to identify that the check is related to that index pattern. I didn't remove it.

File: public/components/health-check/services/check-alerts-pattern.service.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+ *
+ * Find more information about this on the LICENSE file.
+ *
+ */
+import { AppState, SavedObject } from '../../../react-services';
+import { getDataPlugin } from '../../../kibana-services';
+import { HEALTH_CHECK } from '../../../../common/constants';
+import { CheckLogger } from '../types/check_logger';
+
+export const checkAlerts = async (appConfig, checkLogger: CheckLogger) => {
```
Comment (frankeros) [inline public/components/health-check/services/check-alerts-pattern.service.ts:19]: I think `checkIndexPatternObject` is a better name for this service and `check-index-pattern-object.service ` for the file

File: public/components/health-check/services/check-pattern.service.ts [hunk 5]
```diff
@@ -11,92 +11,24 @@
...
-    } else {
-      checkLogger.error('The selected index-pattern is not present');
-    }
+import {
+  checkAlerts,
+  checkFieldsService,
+  checkTemplateService
+} from './index';
+
+export const checkPatternService = (appConfig) => async (checkLogger: CheckLogger) =>  await checkPattern(appConfig, checkLogger);
```
Comment (frankeros) [inline public/components/health-check/services/check-pattern.service.ts:22]: I think `checkIndexPattern` is a better name for this service and `check-index-pattern.service ` for the file

**Review by frankeros — CHANGES_REQUESTED** [review]
Some comments

File: public/components/health-check/container/health-check.container.tsx (hunk 1 above)
Comment (Machi3mfl) [reply public/components/health-check/container/health-check.container.tsx]: Removed in last commit

File: public/components/health-check/container/health-check.container.tsx (hunk 2 above)
Comment (Machi3mfl) [reply public/components/health-check/container/health-check.container.tsx]: Removed in last commit

File: public/components/health-check/container/health-check.container.tsx (hunk 3 above)
Comment (Machi3mfl) [reply public/components/health-check/container/health-check.container.tsx]: Added Alerts in title and label

File: public/components/health-check/services/check-alerts-pattern.service.ts (hunk 4 above)
Comment (Machi3mfl) [reply public/components/health-check/services/check-alerts-pattern.service.ts]: Changed

File: public/components/health-check/services/check-pattern.service.ts (hunk 5 above)
Comment (Machi3mfl) [reply public/components/health-check/services/check-pattern.service.ts]: Changed

**Review by pablomarga — APPROVED** [review]
CR: :heavy_check_mark: 
Testing: :heavy_check_mark: 
Good job!

**Review by gabiwassan — APPROVED** [review]
LGTM! 👍🏼

**Pull request merged**